packages/flutter_tools/test/integration.shard/test_data/migrate_project.dart

```diff
@@ -0,0 +1,227 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@Timeout(Duration(seconds: 600))
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+import '../../src/common.dart';
+import '../test_utils.dart';
+import 'project.dart';
+
+class MigrateProject extends Project {
+  MigrateProject(this.version, {this.vanilla = true});
+
+  @override
+  Future<void> setUpIn(Directory dir, {
+    bool useDeferredLoading = false,
+    bool useSyntheticPackage = false,
+  }) async {
+    this.dir = dir;
+    _appPath = dir.path;
+    if (androidLocalProperties != null) {
+      writeFile(fileSystem.path.join(dir.path, 'android', 'local.properties'), androidLocalProperties);
+    }
+    final Directory tempDir = createResolvedTempDirectorySync('cipd_dest.');
+    final Directory depotToolsDir = createResolvedTempDirectorySync('depot_tools.');
+
+    await processManager.run(<String>[
+      'git',
+      'clone',
+      'https://chromium.googlesource.com/chromium/tools/depot_tools',
+      depotToolsDir.path,
+    ], workingDirectory: dir.path);
+
+    final File cipdFile = depotToolsDir.childFile(globals.platform.isWindows ? 'cipd.bat' : 'cipd');
+    await processManager.run(<String>[
+      cipdFile.path,
+      'init',
+      tempDir.path,
+      '-force',
+    ], workingDirectory: dir.path);
+
+    await processManager.run(<String>[
+      cipdFile.path,
+      'install',
+      'flutter/test/full_app_fixtures/vanilla',
+      version,
+      '-root',
+      tempDir.path,
+    ], workingDirectory: dir.path);
```

Comment:
yeah, if a test requires a cipd dependency, it should be specified in the ci.yaml: https://github.com/flutter/flutter/blob/master/.ci.yaml#L174

Reply:
The way this has to be tested requires frequent setup and teardown of a sample app, and the recipes+deps system is not designed to support this case efficiently. I would need to frequently re-get the cipd deps and the recipes gets deps once at the beginning.

I could host the vanilla sample apps somewhere else and avoid CIPD. The choice to use CIPD is to avoid having to check in entire vanilla apps as test fixtures.

Reply:
We already check in quite a few test apps, I say we just check it in. This way also if the app needs to be fixed, it's easy to find.

Reply:
I checked in a test fixture app for multidex which is a much smaller mini-app, and even that needed to get hosted files such as icon images and stuff.

The test apps required for this must be full apps, which can get quite large. Also, due to the nature of this feature, I intend to eventually branch out to test between many full apps created by different historic flutter versions. Checking in multiple full flutter apps is not sustainable, which is why I turned to hosting on cipd. I could shift the host to github if that makes more sense. I'm not sure if github will appreciate the test infra traffic though, and git makes it much harder to cherrypick individual app fixtures by version and would likely need to be in separate repos to be efficient.

Reply:
Sorry, I missed this message. I'll file a tracking issue for discussion on the approach to this.